beetsplug/parentwork.py

```diff
@@ -0,0 +1,225 @@
+    return work_id, work_date
+
+
+def find_parentwork(work_id):
+    """This function gives the work relationships (dict) of a parent_work
+    given the id of the work"""
+    parent_id, work_date = work_parent(work_id)
+    work_info = musicbrainzngs.get_work_by_id(parent_id,
+                                              includes=["artist-rels"])
+    return work_info, work_date
+
+
+class ParentWorkPlugin(BeetsPlugin):
+    def __init__(self):
+        super(ParentWorkPlugin, self).__init__()
+
+        self.config.add({
+            'auto': False,
+            'force': False,
+        })
+
+        self._command = ui.Subcommand(
+            'parentwork',
+            help=u'Fetches parent works, composers and dates')
+
+        self._command.parser.add_option(
+            u'-f', u'--force', dest='force',
+            action='store_true', default=None,
+            help=u'Re-fetches all parent works')
+
+        if self.config['auto']:
+            self.import_stages = [self.imported]
+
+    def commands(self):
+
+        def func(lib, opts, args):
+            self.config.set_args(opts)
+            force_parent = self.config['force'].get(bool)
+            write = ui.should_write()
+
+            for item in lib.items(ui.decargs(args)):
+                self.find_work(item, force_parent)
+                item.store()
+                if write:
+                    item.try_write()
+
+        self._command.func = func
+        return [self._command]
+
+    def imported(self, session, task):
+        """Import hook for fetching parent works automatically.
+        """
+        force_parent = self.config['force'].get(bool)
+
+        for item in task.imported_items():
+            self.find_work(item, force_parent)
+            item.store()
+
+    def get_info(self, item, work_info, parent_composer, parent_composer_sort,
+                 parent_work, parent_work_disambig, parent_work_id,
+                 composer_ids):
+        """Given the parentwork info dict, this function updates
+        parent_composer, parent_composer_sort, parent_work,
+        parent_work_disambig, work_ids and composer_ids"""
+        composer_exists = False
+        if 'artist-relation-list' in work_info['work']:
+            for artist in work_info['work']['artist-relation-list']:
+                if artist['type'] == 'composer':
+                    composer_exists = True
+                    if artist['artist']['id'] not in composer_ids:
+                        composer_ids.add(artist['artist']['id'])
+                        parent_composer.append(artist['artist']['name'])
+                        parent_composer_sort.append(artist['artist']
+                                                    ['sort-name'])
+        if not composer_exists:
+            self._log.info(item.artist + ' - ' + item.title)
+            self._log.info(
+                "no composer, add one at https://musicbrainz.org/work/" +
+                work_info['work']['id'])
```

Comment:
I don't think we need this log message—if anything, this can go to the debug log.